### PR TITLE
[RFR] rhevm export cleanup

### PR DESCRIPTION
**Pull request by mshriver**
Taken from Amogh's PR #3863

# Testing
I have tested the script, executing locally. It successfully cleaned up old templates on rhevm36, and found no templates to delete on rhvqe40

### CODE REVIEW

File: scripts/cleanup_edomain_templates.py [hunk 1]
```diff
@@ -73,13 +75,16 @@ def change_edomain_state(api, state, edomain):
                 elif state == 'active' and export_domain.get_status().state != 'active':
                     dc.storagedomains.get(edomain).activate()
 
-                wait_for(is_edomain_template_deleted,
-                         [api, state, edomain], fail_condition=False, delay=5)
-                print('{} successfully set to {} state'.format(edomain, state))
+                wait_for(is_edomain_in_state,
+                        [api, state, edomain], fail_condition=False, delay=5)
```
Comment (psav) [inline scripts/cleanup_edomain_templates.py]: Curious as to why we need to pass in api here. Can wrapanapi not be extended to include this function?

File: scripts/cleanup_edomain_templates.py [hunk 2]
```diff
@@ -100,25 +105,36 @@ def get_edomain_path(api, edomain):
...
+        provider_obj: provider object under execution
     """
     try:
-        print("RHEVM: Deleting the empty directories on edomain/vms file...")
-        ssh_client = make_ssh_client(edomainip, sshname, sshpass)
-        command = 'cd {}/master/vms && find . -maxdepth 1 -type d -empty -delete'.format(path)
+        provider_ip = provider_obj.kwargs.get('ipaddress', None)
+        provider_name = provider_obj.kwargs.get('name', None)
+        ssh_client = make_ssh_client(provider_ip, sshname, sshpass)
+        edomain_path = edomainip + ':' + path
```
Comment (psav) [inline scripts/cleanup_edomain_templates.py]: I'd prefer ```"{}:{}".format(...```

File: scripts/cleanup_edomain_templates.py [hunk 3]
```diff
@@ -100,25 +105,36 @@ def get_edomain_path(api, edomain):
...
         sshpass: edomain ssh credentials.
+        provider_obj: provider object under execution
     """
     try:
-        print("RHEVM: Deleting the empty directories on edomain/vms file...")
-        ssh_client = make_ssh_client(edomainip, sshname, sshpass)
-        command = 'cd {}/master/vms && find . -maxdepth 1 -type d -empty -delete'.format(path)
+        provider_ip = provider_obj.kwargs.get('ipaddress', None)
+        provider_name = provider_obj.kwargs.get('name', None)
+        ssh_client = make_ssh_client(provider_ip, sshname, sshpass)
```
Comment (psav) [inline scripts/cleanup_edomain_templates.py]: Shouldn't we already have access to ssh_client for a provider under ```provider_obj.ssh_client``` or is this object just a dict? if it is just the dict of data, we should rename it to provider_dict.

File: scripts/cleanup_edomain_templates.py [hunk 4]
```diff
@@ -278,22 +294,22 @@ def run(**kwargs):
             if args.provider != provider:
                 continue
 
-        mgmt_sys = cfme_data['management_systems'][provider]
-        ssh_rhevm_creds = mgmt_sys['hosts'][0]['credentials']
+        provider_obj = get_mgmt(provider)
```
Comment (psav) [inline scripts/cleanup_edomain_templates.py]: Same here, this is misleading. We should call it ```provider_mgmt```

**Comment by mshriver** [comment]
@psav I'm looking into your comments, ripped this code from Amogh's branch.

File: scripts/cleanup_edomain_templates.py (hunk 3 above)
Comment (mshriver) [inline scripts/cleanup_edomain_templates.py]: Its an object, but there is no ssh_client attribute. ssh_client is defined for utils.appliance and utils.composite but not in the provider classes.

File: scripts/cleanup_edomain_templates.py (hunk 2 above)
Comment (mshriver) [inline scripts/cleanup_edomain_templates.py]: agreed

**Comment by mshriver** [comment]
@dajohnso 

Script is working from local branch, would like to get merged so Jenkins picks it up and starts maintaining rhevm edomain again.

**Pull request merged**